lib/Models/Terria.ts

```diff
@@ -1241,6 +1241,49 @@ export default class Terria {
     );
   }
 
+  // This method is created to support unit tests.
+  async loadMapItems(model: BaseModel, errors: TerriaError[]) {
+    if (MappableMixin.isMixedInto(model)) {
+      (await model.loadMapItems()).pushErrorTo(errors);
+    }
+  }
+
+  private async pushAndLoadMapItems(
+    model: BaseModel,
+    newItems: BaseModel[],
+    errors: TerriaError[]
+  ) {
+    if (ReferenceMixin.isMixedInto(model)) {
+      (await model.loadReference()).pushErrorTo(errors);
+
+      if (model.target !== undefined) {
+        await this.pushAndLoadMapItems(model.target, newItems, errors);
+      } else {
+        errors.push(
+          TerriaError.from(
+            "Reference model has no target. Model Id: " + model.uniqueId
+          )
+        );
+      }
+    } else if (GroupMixin.isMixedInto(model)) {
+      (await model.loadMembers()).pushErrorTo(errors);
+
+      model.memberModels.map(async m => {
+        await this.pushAndLoadMapItems(m, newItems, errors);
+      });
+    } else if (MappableMixin.isMixedInto(model)) {
+      newItems.push(model);
+      await this.loadMapItems(model, errors);
+    } else {
+      errors.push(
+        TerriaError.from(
+          "Can not load an un-mappable item to the map. Item Id: " +
+            model.uniqueId
+        )
+      );
+    }
+  }
```

Comment:
These shouldn't be in Terria.ts. Better places for these would be named exports from other files.

* `loadMapItems` could be exported from MappableMixin as a separate named export
* `pushAndLoadMapItems` could be its own file

Reply:
`loadmapItems` is only a wrapper -- convenient for unit test to spy on.